### PR TITLE
import wallet with seedphrases on qr screen

### PR DESCRIPTION
**Pull request by Kamill90**

### CODE REVIEW

File: src/screens/ImportWalletQRCodeScreen.tsx [hunk 1]
```diff
@@ -65,220 +69,396 @@ export default class ImportWalletQRCodeScreen extends React.Component<Props, Sta
...
+      title: i18n.message.creatingWallet,
+      description: i18n.message.creatingWalletDescription,
+      type: MessageType.processingState,
+      asyncTask: () => this.importMnemonic(event.data),
+    });
+    this.props.navigation.goBack();
+
+    // code for importing wallet only with qr code with bug descripted in ticket https://trello.com/c/5TgN4PqE/193-import-walletu-android-iphone
+
+    // if (now() - this.lastTimeIveBeenHere < SCAN_CODE_AFTER_MS) {
```
Comment (bklukaczewski) [inline src/screens/ImportWalletQRCodeScreen.tsx:106]: can we remove it? maybe in separate commit, so it's easily revertible?

File: src/screens/ImportWalletQRCodeScreen.tsx [hunk 2]
```diff
@@ -65,220 +69,396 @@ export default class ImportWalletQRCodeScreen extends React.Component<Props, Sta
...
     }
+  };
 
-    for (const w of BlueApp.wallets) {
-      if (w.getSecret() === event.data) {
-        // lookig for duplicates
-        this.setState({ isLoading: false });
-        return Alert.alert(loc.wallets.scanQrWif.wallet_already_exists); // duplicate, not adding
+  importMnemonic = async (text: string) => {
+    console.log('importMnemonic');
```
Comment (bklukaczewski) [inline src/screens/ImportWalletQRCodeScreen.tsx:337]: Would be great to gradually remove all console logs as they hurt performance in production

File: src/screens/ImportWalletQRCodeScreen.tsx [hunk 3]
```diff
@@ -65,220 +69,396 @@ export default class ImportWalletQRCodeScreen extends React.Component<Props, Sta
...
-      this.props.navigation.popToTop();
-      setTimeout(() => EV(EV.enum.WALLETS_COUNT_CHANGED), 500);
-      return;
-    }
+      const watchOnly = new WatchOnlyWallet();
+      watchOnly.setSecret(text);
+      if (watchOnly.valid()) {
+        await watchOnly.fetchTransactions();
+        await watchOnly.fetchBalance();
+        return this.saveWallet(watchOnly);
```
Comment (bklukaczewski) [inline src/screens/ImportWalletQRCodeScreen.tsx:439]: Huge kudos for this refactoring 💪

File: src/screens/SendCoinsScreen.tsx [hunk 4]
```diff
@@ -440,7 +440,6 @@ export class SendCoinsScreen extends Component<Props, State> {
           console.log('try #', tries, 'fee=', fee);
           if (this.recalculateAvailableBalance(this.state.fromWallet.getBalance(), firstTransaction.amount, fee) < 0) {
             // we could not add any fee. user is trying to send all he's got. that wont work
-            Alert.alert('wtf');
```
Comment (bklukaczewski) [inline src/screens/SendCoinsScreen.tsx:443]: 😀

File: src/screens/ImportWalletQRCodeScreen.tsx (hunk 2 above)
Comment (Kamill90) [reply src/screens/ImportWalletQRCodeScreen.tsx:337]: Agree, console.logs should be treated as errors by eslint

Comment (bklukaczewski) [reply src/screens/ImportWalletQRCodeScreen.tsx]: https://trello.com/c/qQQ43w0P/218-add-eslint-rule-for-consolelogs I've created a task for that

**Pull request merged**